src/map/clif.c

```diff
@@ -267,19 +267,39 @@ uint32 clif_refresh_ip(void)
 
 #if PACKETVER >= 20071106
 static inline unsigned char clif_bl_type(struct block_list *bl) {
+	struct view_data *vd;
 	nullpo_retr(0x1, bl);
+
+	vd = status->get_viewdata(bl);
```

Comment:
we can't do a nullpo check on vd as it's not always used and maybe can be null in cases it's not used, check rest of he code.